### PR TITLE
Fixed login/logout on management page

### PR DESCRIPTION
**Pull request by KostashchukIryna**
### Issue Link 📋
https://github.com/ita-social-projects/GreenCity/issues/7999
https://github.com/ita-social-projects/GreenCity/issues/8005

### Summary Of Changes 

- Login button is now responsive, 
- added logout button, 
- About Us page is now accessible to unauthorized/unathenticated users with ability to be redirected to login   

**After:**

https://github.com/user-attachments/assets/a262909a-05ee-42b9-8d69-c91642b59ea3

## Release Notes

- **New Features**
  - Introduced a new `SecurityUtils` utility class for user authentication checks.
  - Added a new constant for representing anonymous users.

- **Improvements**
  - Simplified authentication logic across multiple controllers.
  - Updated navigation endpoints for About Us and Management pages.
  - Enhanced login/logout functionality in the header template.

- **Bug Fixes**
  - Improved user authentication detection mechanism.

- **Documentation**
  - Added comprehensive unit tests for `SecurityUtils`.

- **Style**
  - Updated CSS for main content layout in the About Us page.

### CODE REVIEW

**Comment by KostashchukIryna** [comment]
@coderabbitai review

File: core/src/main/resources/templates/core/header.html [hunk 1]
```diff
@@ -20,7 +20,12 @@
                         </div>
                     </div>
                     <div class="login">
-                        <button formaction="/management/login">Login</button>
+                        <form th:if="${T(greencity.security.utils.SecurityUtils).isAuthenticated()}" action="/management/logout" method="get" >
+                            <button type="submit">Logout</button>
+                        </form>
```
[the remark this answers is not in the document]
Comment (KostashchukIryna) [reply core/src/main/resources/templates/core/header.html:25]: "/management/logout", "GET" is in SecurityConfig

File: core/src/main/java/greencity/security/utils/SecurityUtils.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.NoArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class SecurityUtils {
+    public static boolean isAuthenticated() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        return authentication != null && authentication.isAuthenticated()
+            && !"anonymousUser".equals(authentication.getPrincipal().toString());
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/security/utils/SecurityUtils.java]: Please move `anonymousUser` to the constant

File: core/src/test/java/greencity/security/utils/SecurityUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    @Test
+    void isAuthenticatedFalseForNullAuthenticationTest() {
+        when(mockSecurityContext.getAuthentication()).thenReturn(null);
+
+        boolean result = SecurityUtils.isAuthenticated();
+        assertFalse(result);
+    }
+
+    @Test
+    void isAuthenticatedFalseForAnonymousUserTest() {
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/security/utils/SecurityUtilsTest.java:79]: please add test if `principal` is `null`

**Review by KizerovDmitriy — APPROVED** [review]

File: core/src/main/java/greencity/security/utils/SecurityUtils.java (hunk 2 above)
Comment (KostashchukIryna) [reply core/src/main/java/greencity/security/utils/SecurityUtils.java]: Thank you, changed that

File: core/src/test/java/greencity/security/utils/SecurityUtilsTest.java (hunk 3 above)
Comment (KostashchukIryna) [reply core/src/test/java/greencity/security/utils/SecurityUtilsTest.java:79]: Thank you, added that

**Review by VasylyshynDmytro — APPROVED** [review]

**Review by VasylyshynDmytro — APPROVED** [review]

**Review by urio999 — APPROVED** [review]
